app/routes/action.js

```diff
@@ -116,5 +126,66 @@ module.exports = [
         })
       return h.response(filteredActions).code(OK_STATUS_CODE)
     }
+  },
+  {
+    method: 'POST',
+    path: '/action/{pathParam}/rule/',
+    options: {
+      validate: {
+        payload: Joi.object({
+          id: Joi.string().required(),
+          config: Joi.object().optional()
+        })
+      }
+    },
+    handler: (request, h) => commonHandler(request, h, (action, request, h) => {
+      const newRule = request.payload
+      addRule(action, newRule)
+      console.log(JSON.stringify(getActions(), null, 2))
+      return h.response({ message: 'Rule added successfully' }).code(OK_STATUS_CODE)
+    })
+  },
+  {
+    method: 'PUT',
+    path: '/action/{pathParam}/rule/',
+    options: {
+      validate: {
+        payload: Joi.object({
+          id: Joi.string().required(),
+          config: Joi.object().optional()
+        })
+      }
+    },
+    handler: (request, h) => commonHandler(request, h, (action, request, h) => {
+      const newRule = request.payload
+      const ruleIndex = findRuleIndex(action.eligibilityRules, newRule.id)
```

Comment:
The logic related to the findRuleIndex invocation can be moved into land-action/index.js.
It isn't a concern or responsibility of the handler function. 
For example, the updateRule function could return undefined or null when an invalid rule id is provided

Reply:
Neat point! Done :D